R/mediate.R

```diff
@@ -0,0 +1,394 @@
+             
+          # Update index 
+          i <- i + 1
+          
+          if (verbose) {
+            message("\rMediator ", i, " out of ", length(mediators), ": ", mediator, appendLF = FALSE) 
+          }
+          
+          # Run mediation analysis for current mediator
+          med_out <- run.mediate(x, outcome, treatment, mediator,
+                                  family = family, mat = mat,
+                                  covariates = covariates, ...)
+          # Update list of results
+          results <- update.results(results, med_out, treatment, mediator, outcome)
+        }
+        
+        # Combine results into dataframe
+        med_df <- make.output(results, p.adj.method, add.metadata)
+        return(med_df)
+    }
+)
+
+
+# Check that arguments can be passed to mediate and remove unused samples
+check.mediate.args <- function(x, outcome, treatment, mediator,
+                               covariates, verbose = TRUE, ...) {
+  
+  # Create dataframe from selected columns of colData
+  df <- as.data.frame(colData(x)[ , names(colData(x)) %in% c(outcome, treatment, mediator, covariates)])
+  # Store kwargs into variable
+  kwargs <- list(...)
+  
+  # Remove missing data from df
+  df <- na.omit(df)
+  diff <- ncol(x) - nrow(df)
+  
+  if (diff != 0) {
+    # Remove missing data from se
+    x <- x[ , rownames(df)]
+    
+    if (verbose) {
+      message(diff, " samples removed because of missing data.")
+    }
+  }
+  
+  # If the treatment variable has three or more levels
+  if (!is.numeric(df[[treatment]]) & length(unique((df[[treatment]]))) > 2) {
+    # and boot is TRUE
+    if (!is.null(kwargs[["boot"]])) {
+      # and control and treat value are not specified
+      if (any(sapply(kwargs[c("control.value", "treat.value")], is.null))) {
+        stop("Too many treatment levels. Consider specifing a treat.value and a control.value", call. = FALSE)
+      # but if they are specified
+      } else {
+        # and they also appear in the treatment variable
+        if (!all(kwargs[c("control.value", "treat.value")] %in% unique(df[[treatment]]))) {
+          stop("treat.value and/or control.value not found in the levels of the treatment variable.", call. = FALSE)
+        }
+        
+        # Find indices of samples that belong to either control or treatment
+        keep <- df[[treatment]] %in% kwargs[c("control.value", "treat.value")]
+        
+        # Remove samples different from control and treatment from df
+        df <- df[keep, ]
+        diff <- ncol(x) - nrow(df)
+        
+        # Remove samples different from control and treatment from se
+        x <- x[ , rownames(df)]
+        
+        if (verbose) {
+          message(diff, " samples removed because different from control and treatment.")
+        }
+      }
+    }
+  }
+  return(x)
+}
+
+# Run mediation analysis
+#' @importFrom mediation mediate
+run.mediate <- function(x, outcome, treatment,
```

Comment:
If this is internal function then you could start it with period, so it would be .run.mediate

Reply:
same for other internals